lib/rubycritic/platforms/base.rb

```diff
@@ -0,0 +1,39 @@
+require "rubycritic/configuration"
+
+module Rubycritic
+  module Platforms
+    class Base
+      attr_reader :browser_name
+
+      def initialize
+        @browser_name = Config.open_with
+      end
+
+      def self.can_open?
+        raise NotImplementedError.new("The #{self.class} class must implement the #{__method__} method.")
+      end
+
+      def browser
+        raise NotImplementedError.new("The #{self.class} class must implement the #{__method__} method.")
+      end
+
+      def open(_)
+        raise NotImplementedError.new("The #{self.class} class must implement the #{__method__} method.")
+      end
+
+      def application
+        raise NotImplementedError.new("The #{self.class} class must implement the #{__method__} method.")
```

Comment:
Those error messages are very repetitive - given that "normal" user never see this message I'd rather go for just

```Ruby
raise NotImplementedError
```
with no explicit error message to cut down repetition and improve readability.